web/src/services/UserService.ts

```diff
@@ -62,3 +61,35 @@ export async function postUser(user: PostUserSchema){
     .then(response => response.json())
     .catch(error => console.error(error));
 }
+
+export async function putUser(id: number, user: UserSchema, action: number) {
+    let response = null;
+
+    try {
+        if (action === 1) {
+            response = await fetch(`${API_URL}/${id}/desactivate`, {
+                method: 'PUT',
+                headers: {
+                    "Content-Type": 'application/json'
+                },
+                body: JSON.stringify(user)
+            });
+        }
+        if (action === 2) {
+            response = await fetch(`${API_URL}/${id}/activate`, {
+                method: 'PUT',
+                headers: {
+                    "Content-Type": 'application/json'
+                },
+                body: JSON.stringify(user)
+            });
+        }
+        if (!response || !response.ok) {
+            throw new Error(`Erro ao atualizar o usuário: ${response ? response.statusText : 'Resposta não recebida'}`);
+        }
+        return response.json();
+    } catch (error) {
+        console.error("Erro ao atualizar o usuário:", error);
+        throw error;
+    }
+}
```

Comment:
a função possui um nome genérico para uma operação específica, o nome `PUT` implica que podemos efetuar qualquer tipo de atualização no usuário sendo que a função é responsável apenas por atualizar o "status ativo" do usuário (ativar / desativar).

aqui está uma sugestão de como refatorar a função para reduzir repetição de código e melhorar a clareza

```suggestion
export async function updateUserActiveStatus(user: UserSchema, newActiveStatus: boolean) {
    let response = null;
    let endpoint = newActiveStatus ? 'activate' : 'deactivate';

    try {
            response = await fetch(`${API_URL}/${user.id}/${endpoint}`, {
                method: 'PUT',
                headers: {
                    "Content-Type": 'application/json'
                },
                body: JSON.stringify(user)
            });
        }
        if (!response || !response.ok) {
            throw new Error(`Erro ao atualizar o usuário: ${response ? response.statusText : 'Resposta não recebida'}`);
        }
        return response.json();
    } catch (error) {
        console.error("Erro ao atualizar o usuário:", error);
        throw error;
    }
}
```